### PR TITLE
Basic signup

### PR DESCRIPTION
**Pull request by freddiehonohan**
Added basic signup endpoint +test, and frontend view which currently performs an auto-login. 

Let me know if you see any changes. I also refactored some of the test scripts, can take them out if they're breaking for you.

### CODE REVIEW

File: {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/tests/test_auth.py [hunk 1]
```diff
@@ -14,6 +14,21 @@ def verify_password_mock(first: str, second: str):
     assert response.status_code == 200
 
 
+def test_signup(client):
```
Comment (Buuntu) [inline {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/tests/test_auth.py:17]: can you monkeypatch the hashing function here like I did in test_login (or verify_password)? it adds about a second or so per test and we don't need to test that it hashes correctly?

Comment (Buuntu) [reply {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/tests/test_auth.py:17]: The already exists test can also use the test_user fixture and be a separate test and I don't think it should add much time.

File: {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/auth.py [hunk 2]
```diff
@@ -34,3 +34,30 @@ async def login(
...
+
+
+@r.post("/signup")
+async def signup(
+    db=Depends(get_db), form_data: OAuth2PasswordRequestForm = Depends()
+):
+    user = sign_up_new_user(db, form_data.username, form_data.password)
+    if not user:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
```
Comment (Buuntu) [inline {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/auth.py:46]: do you think 403 is the right status code for things that already exist? is that what FastAPI recommends?

File: {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/tests/test_auth.py [hunk 3]
```diff
@@ -14,6 +14,21 @@ def verify_password_mock(first: str, second: str):
...
+    response = client.post(
+        "/api/signup",
+        data={"username": 'some@email.com', "password": "randompassword"},
+    )
+    assert response.status_code == 200
+
+    # Try again with same credentials
+    response = client.post(
```
Comment (Buuntu) [inline {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/tests/test_auth.py:25]: The already exists test can also use the `test_user` fixture and be a separate test and I don't think it should add much time.

File: {{cookiecutter.project_slug}}/scripts/build.sh [hunk 4]
```diff
@@ -4,7 +4,7 @@
 docker-compose up -d
 
 # Hack to wait for postgres container to be up before running alembic migrations
-sleep 3;
+sleep 5;
```
[the remark this answers is not in the document]
Comment (freddiehonohan) [reply {{cookiecutter.project_slug}}/scripts/build.sh:7]: Yes, this sleep works 100% of the time on my 5 year old laptop

[the remark this answers is not in the document]
Comment (Buuntu) [reply {{cookiecutter.project_slug}}/scripts/build.sh:7]: There is probably a better way to handle this but I'll look into it

File: {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/tests/test_auth.py [hunk 5]
```diff
@@ -14,6 +15,25 @@ def verify_password_mock(first: str, second: str):
     assert response.status_code == 200
 
 
+def test_signup(client):
+    response = client.post(
```
Comment (Buuntu) [inline {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/tests/test_auth.py:19]: This should also use a monkeypatch, the example I have above should work. In this case it's a different function though (get_password_hash)

File: {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/auth.py (hunk 2 above)
Comment (Buuntu) [reply {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/auth.py]: I often see 422 as a catch all for validation errors, 409 is fine too you just don't see it as often.

Comment (freddiehonohan) [reply {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/auth.py]: I'll change it so, along with the monkeypatch :+1:

Comment (freddiehonohan) [reply {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/auth.py]: HTTP_422_UNPROCESSABLE_ENTITY

Comment (Buuntu) [reply {{cookiecutter.project_slug}}/backend/app/api/api_v1/routers/auth.py]: Okay, that's fine.

**Pull request merged**